### PR TITLE
port: fix issues w/ ExpectReplies, invoke in SkillDialog

### LINKED ISSUE #2962: Port: Fix issues with ExpectReplies and Invoke in SkillDialog
While adding BufferedReplies, type of InvokeResponse should be added to TurnState:

```cs
                        if (activity.Type == ActivityTypesEx.InvokeResponse)
                        {
                            TurnState.Add(BotFrameworkAdapter.InvokeResponseKey, activity);
                        }
```

https://github.com/microsoft/botbuilder-dotnet/pull/4845

And, on the Skill Dialog side, only one InvokeResponse should be sent.

### PR DESCRIPTION
**Pull request by joshgummersall**
Also includes some small cleanup/formatting changes.

Fixes #2962

### CODE REVIEW

File: libraries/botbuilder-core/src/skills/botFrameworkClient.ts [hunk 1]
```diff
@@ -21,21 +21,12 @@ export interface BotFrameworkClient {
...
-          ) => Promise<InvokeResponse<T>>)
-        | ((
-              fromBotId: string,
-              toBotId: string,
-              toUrl: string,
-              serviceUrl: string,
-              conversationId: string,
-              activity: Activity
-          ) => Promise<InvokeResponse>);
+    postActivity: <T = any>(
```
Comment (joshgummersall) [inline libraries/botbuilder-core/src/skills/botFrameworkClient.ts:24]: This is the fix to remove the unnecessary `as` casting [here](https://github.com/microsoft/botbuilder-js/pull/2966/files#diff-6c0e837cf326652d3b36fad7a83a81d94cacee1e8835f2f535e017d7d83876adL269). The type signatures are equivalent.

File: libraries/botbuilder-core/src/turnContext.ts [hunk 2]
```diff
@@ -483,29 +484,40 @@ export class TurnContext {
...
+                this.bufferedReplyActivities.push(...outputActivities);
+                const responses = outputActivities.map<ResourceResponse>(() => ({ id: undefined }));
+
+                // Ensure the TurnState has the InvokeResponseKey, since this activity
+                // is not being sent through the adapter, where it would be added to TurnState.
+                const invokeResponse = outputActivities.find(
+                    (activity) => activity.type === ActivityTypes.InvokeResponse
+                );
+                if (invokeResponse) {
+                    this.turnState.set(INVOKE_RESPONSE_KEY, invokeResponse);
```
Comment (joshgummersall) [inline libraries/botbuilder-core/src/turnContext.ts:519]: @stevengum and/or @EricDahlvang I'm pretty sure this is roughly equivalent to https://github.com/microsoft/botbuilder-dotnet/pull/4845/files#diff-391e4084826390583f2af2b2d27d0d0add457734d88f05e817e826af88a2b41aR328.

File: libraries/botframework-schema/src/index.ts [hunk 3]
```diff
@@ -1753,6 +1753,7 @@ export enum ActivityTypes {
   EndOfConversation = 'endOfConversation',
   Event = 'event',
   Invoke = 'invoke',
+  InvokeResponse = 'invokeResponse',
```
Comment (joshgummersall) [inline libraries/botframework-schema/src/index.ts:1756]: I did not see an `ActivityTypesEx` enum in the JS SDK so I just added this to the existing `ActivityTypes` enum.

File: libraries/botbuilder-core/src/turnContext.ts [hunk 4]
```diff
@@ -503,8 +504,15 @@ export class TurnContext {
                 // Append activities to buffer
                 const responses: ResourceResponse[] = [];
                 output.forEach((a) => {
-                        this.bufferedReplyActivities.push(a);
-                        responses.push({ id: undefined });
+                    this.bufferedReplyActivities.push(a);
+
+                    // Ensure the TurnState has the InvokeResponseKey, since this activity
+                    // is not being sent through the adapter, where it would be added to TurnState.
+                    if (a.type === ActivityTypes.InvokeResponse) {
```
Comment (joshgummersall) [inline libraries/botbuilder-core/src/turnContext.ts:511]: This is, I think, equivalent to https://github.com/microsoft/botbuilder-dotnet/pull/4845/files#diff-391e4084826390583f2af2b2d27d0d0add457734d88f05e817e826af88a2b41aR328

File: libraries/botframework-schema/src/index.ts (hunk 3 above)
Comment (gabog) [reply libraries/botframework-schema/src/index.ts:1756]: That's OK, ActivityTypesEx was created when we used to create the Activity and other classes from swagger, we don't do that anymore.

**Review by gabog — APPROVED** [review]
Looks good to me, @EricDahlvang, would you be able to take a quick pass and make sure nothing is missing?

**Review by EricDahlvang — APPROVED** [review]
lgtm

File: libraries/botbuilder-core/src/turnContext.ts (hunk 4 above)
Comment (stevengum) [reply libraries/botbuilder-core/src/turnContext.ts:511]: The JS code looks good 👍 

@gabog, @johnataylor do we need to do a TryAdd in .NET or some sort of TrySet?

Comment (stevengum) [reply libraries/botbuilder-core/src/turnContext.ts:511]: We want to do a last-write wins in .NET if I'm correctly recall the implementation around TeamsActivityHandler. `.Add()` (depending on the class/implementation) normally throws an exception iirc.

**Review by stevengum — APPROVED** [review]
This looks fine... Are we squared away in .NET with test coverage? If not can you do a pass at testing the JS E2E with @gabog's assistance and make sure test coverage is added for these changes in JS? 

(Either update the PR with some unit tests (preferable) or file an issue for R12 to add appropriate unit tests)

**Comment by joshgummersall** [comment]
I've created https://github.com/microsoft/botbuilder-js/issues/2969 so we don't hold up the new RC.

**Pull request merged**